core/src/main/java/nl/nn/adapterframework/http/WebServiceListener.java

```diff
@@ -137,16 +137,19 @@ public void open() throws ListenerException {
 			} else {
 				log.error("unable to publish listener ["+getName()+"] on CXF endpoint ["+getAddress()+"]");
 			}
-		}
 
-		// Register with dispatcher so it can be used on the console 'Test ServiceListener' page.
-		if (StringUtils.isNotEmpty(getServiceNamespaceURI())) {
-			log.debug("registering listener ["+getName()+"] with ServiceDispatcher by serviceNamespaceURI ["+getServiceNamespaceURI()+"]");
+			// Register with dispatcher so it can be used on the console 'Test ServiceListener' page.
+			log.debug("registering listener ["+getName()+"] with ServiceDispatcher by address ["+getAddress()+"]");
 			ServiceDispatcher.getInstance().registerServiceClient(getServiceNamespaceURI(), this);
-		}
-		else {
-			log.debug("registering listener ["+getName()+"] with ServiceDispatcher");
-			ServiceDispatcher.getInstance().registerServiceClient(getName(), this); //Backwards compatibility + Console 'Test ServiceListener'
+		} else {
+			if (StringUtils.isNotEmpty(getServiceNamespaceURI())) {
```

Comment:
Je zou nog deze twee regels kunnen mergen in een else-if, geloof ik?

Reply:
Ja, ik ben al lang blij als het weer werkt :( in de master veel aangepast...dat wel